presto-main/src/main/java/io/prestosql/sql/planner/RelationPlanner.java

```diff
@@ -238,17 +239,21 @@ private RelationPlan addColumnMasks(Table table, RelationPlan plan)
                 for (Symbol symbol : plan.getRoot().getOutputSymbols()) {
                     assignments.put(symbol, symbol.toSymbolReference());
                 }
-                assignments.put(plan.getFieldMappings().get(i), coerceIfNecessary(analysis, mask, planBuilder.rewrite(mask)));
+
+                Symbol symbol = symbolAllocator.newSymbol(plan.getFieldMappings().get(i));
+                assignments.put(symbol, coerceIfNecessary(analysis, mask, planBuilder.rewrite(mask)));
+                newFieldMappings.set(i, symbol);
 
                 planBuilder = planBuilder
```

Comment:
why `plan` is not update here (after `ProjectNode` is added? Specifically we do loop:
```
                for (Symbol symbol : plan.getRoot().getOutputSymbols()) {
                    assignments.put(symbol, symbol.toSymbolReference());
                }
```
but it seems to be missing masked symbols added by previous `ProjectNode`

Reply:
Huh, yeah, that looks like a bug. It should be using planBuilder.getRoot(). I’ll add a test